### ISSUE TITLE
SwiftUI Code does not work within NavigationView

### ISSUE THREAD

**Issue by muthenberg** [hCaptcha/HCaptcha-ios-sdk#91]
The Swift UI Example from the documentation works just fine, as long as you have not included the code into a NavigationView. If the code is used as the second view in a navigation view, as soon as you hit 'validate', the challenge is issued, but not shown on the screen. After some time, the challenge resets and finally is shown, but it takes about a minute or longer, which is too long for my users to wait.

Maybe I am doing something wrong here. Hints would be much appreciated. On the other hand, maybe the SwiftUI example is too simplistic and should be updated with further details.

I am too unexperienced to see what the problem is, but I've created a [reproducer](https://github.com/muthenberg/hcaptcha-swiftui-reproducer) and I got the log output, which reads:

```
=== AttributeGraph: cycle detected through attribute 7744 ===
2022-07-22 08:50:26.703200+0200 HCaptchaTest[77715:5241039] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x60000062ed50 'BIB_Leading_Leading' H:|-(0)-[_UIModernBarButton:0x150679bf0]   (active, names: '|':_UIButtonBarButton:0x15211f930 )>",
    "<NSLayoutConstraint:0x60000062fac0 'UINav_static_button_horiz_position' _UIModernBarButton:0x150679bf0.leading == UILayoutGuide:0x600001c15b20'UIViewLayoutMarginsGuide'.leading   (active)>",
    "<NSLayoutConstraint:0x60000062fb10 'UINavItemContentGuide-leading' H:[_UIButtonBarButton:0x15211f930]-(6)-[UILayoutGuide:0x600001c15a40'UINavigationBarItemContentLayoutGuide']   (active)>",
    "<NSLayoutConstraint:0x600000658e60 'UINavItemContentGuide-trailing' UILayoutGuide:0x600001c15a40'UINavigationBarItemContentLayoutGuide'.trailing == _UINavigationBarContentView:0x15211a4b0.trailing   (active)>",
    "<NSLayoutConstraint:0x6000006251d0 'UIView-Encapsulated-Layout-Width' _UINavigationBarContentView:0x15211a4b0.width == 0   (active)>",
    "<NSLayoutConstraint:0x600000659180 'UIView-leftMargin-guide-constraint' H:|-(8)-[UILayoutGuide:0x600001c15b20'UIViewLayoutMarginsGuide'](LTR)   (active, names: '|':_UINavigationBarContentView:0x15211a4b0 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000062ed50 'BIB_Leading_Leading' H:|-(0)-[_UIModernBarButton:0x150679bf0]   (active, names: '|':_UIButtonBarButton:0x15211f930 )>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.
2022-07-22 08:50:26.703626+0200 HCaptchaTest[77715:5241039] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x60000062eda0 'BIB_Trailing_CB_Leading' H:[_UIModernBarButton:0x150679bf0]-(6)-[_UIModernBarButton:0x15211fd10'Main Page']   (active)>",
    "<NSLayoutConstraint:0x60000062edf0 'CB_Trailing_Trailing' _UIModernBarButton:0x15211fd10'Main Page'.trailing <= _UIButtonBarButton:0x15211f930.trailing   (active)>",
    "<NSLayoutConstraint:0x60000062fac0 'UINav_static_button_horiz_position' _UIModernBarButton:0x150679bf0.leading == UILayoutGuide:0x600001c15b20'UIViewLayoutMarginsGuide'.leading   (active)>",
    "<NSLayoutConstraint:0x60000062fb10 'UINavItemContentGuide-leading' H:[_UIButtonBarButton:0x15211f930]-(6)-[UILayoutGuide:0x600001c15a40'UINavigationBarItemContentLayoutGuide']   (active)>",
    "<NSLayoutConstraint:0x600000658e60 'UINavItemContentGuide-trailing' UILayoutGuide:0x600001c15a40'UINavigationBarItemContentLayoutGuide'.trailing == _UINavigationBarContentView:0x15211a4b0.trailing   (active)>",
    "<NSLayoutConstraint:0x6000006251d0 'UIView-Encapsulated-Layout-Width' _UINavigationBarContentView:0x15211a4b0.width == 0   (active)>",
    "<NSLayoutConstraint:0x600000659180 'UIView-leftMargin-guide-constraint' H:|-(8)-[UILayoutGuide:0x600001c15b20'UIViewLayoutMarginsGuide'](LTR)   (active, names: '|':_UINavigationBarContentView:0x15211a4b0 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000062eda0 'BIB_Trailing_CB_Leading' H:[_UIModernBarButton:0x150679bf0]-(6)-[_UIModernBarButton:0x15211fd10'Main Page']   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.
2022-07-22 08:50:26.704075+0200 HCaptchaTest[77715:5241039] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x60000062e2b0 UIView:0x15067c2d0.trailing == _UIBackButtonMaskView:0x15067bb70.trailing   (active)>",
    "<NSLayoutConstraint:0x60000062f390 'Mask_Trailing_Trailing' _UIBackButtonMaskView:0x15067bb70.trailing == _UIButtonBarButton:0x15211f930.trailing   (active)>",
    "<NSLayoutConstraint:0x60000062f4d0 'MaskEV_Leading_BIB_Trailing' H:[_UIModernBarButton:0x150679bf0]-(0)-[UIView:0x15067c2d0]   (active)>",
    "<NSLayoutConstraint:0x60000062fac0 'UINav_static_button_horiz_position' _UIModernBarButton:0x150679bf0.leading == UILayoutGuide:0x600001c15b20'UIViewLayoutMarginsGuide'.leading   (active)>",
    "<NSLayoutConstraint:0x60000062fb10 'UINavItemContentGuide-leading' H:[_UIButtonBarButton:0x15211f930]-(6)-[UILayoutGuide:0x600001c15a40'UINavigationBarItemContentLayoutGuide']   (active)>",
    "<NSLayoutConstraint:0x600000658e60 'UINavItemContentGuide-trailing' UILayoutGuide:0x600001c15a40'UINavigationBarItemContentLayoutGuide'.trailing == _UINavigationBarContentView:0x15211a4b0.trailing   (active)>",
    "<NSLayoutConstraint:0x6000006251d0 'UIView-Encapsulated-Layout-Width' _UINavigationBarContentView:0x15211a4b0.width == 0   (active)>",
    "<NSLayoutConstraint:0x600000659180 'UIView-leftMargin-guide-constraint' H:|-(8)-[UILayoutGuide:0x600001c15b20'UIViewLayoutMarginsGuide'](LTR)   (active, names: '|':_UINavigationBarContentView:0x15211a4b0 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000062e2b0 UIView:0x15067c2d0.trailing == _UIBackButtonMaskView:0x15067bb70.trailing   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.
[JS LOG]: recaptchacompat disabled
[JS LOG]: challenge onload starting
2022-07-22 08:50:26.921469+0200 HCaptchaTest[77715:5241039] [ViewportSizing] maximumViewportInset cannot be larger than frame
2022-07-22 08:50:26.921518+0200 HCaptchaTest[77715:5241039] [ViewportSizing] minimumViewportInset cannot be larger than frame
2022-07-22 08:50:26.921705+0200 HCaptchaTest[77715:5241039] [ViewportSizing] maximumViewportInset cannot be larger than frame
2022-07-22 08:50:26.921735+0200 HCaptchaTest[77715:5241039] [ViewportSizing] minimumViewportInset cannot be larger than frame
2022-07-22 08:50:30.037883+0200 HCaptchaTest[77715:5241039] [ViewportSizing] maximumViewportInset cannot be larger than frame
2022-07-22 08:50:30.038063+0200 HCaptchaTest[77715:5241039] [ViewportSizing] minimumViewportInset cannot be larger than frame
[JS LOG]: challenge executing
[JS LOG]: showing challenge
[JS LOG]: challenge opened
2022-07-22 08:50:31.845902+0200 HCaptchaTest[77715:5241039] [HardwareKeyboard] -[UIApplication getKeyboardDevicePropertiesForSenderID:shouldUpdate:usingSyntheticEvent:], failed to fetch device property for senderID (778835616971358211) use primary keyboard info instead.
2022-07-22 08:50:31.847408+0200 HCaptchaTest[77715:5241039] [HardwareKeyboard] -[UIApplication getKeyboardDevicePropertiesForSenderID:shouldUpdate:usingSyntheticEvent:], failed to fetch device property for senderID (778835616971358211) use primary keyboard info instead.
[JS LOG]: expired challenge
[JS LOG]: challenge resetting
[JS LOG]: challenge executing
[JS LOG]: showing challenge
[JS LOG]: challenge opened
```

**Comment by CAMOBAP** [comment]
It looks like our example isn't quite right and we have a race in it. The problem happens because SDK fire `configureWebView` during initialization and SwiftUI may not finish "Layout phase" during this i.e. `self.placeholder.uiview.bounds` is zero

@e271828- I think we can fix this on SDK size to call configure one more time if `size == CGSize.zero`. Also, make sens to add SwiftUI example as actual target to our project (so we will be able to test it on each changes)

```diff
diff --git a/HCaptcha/Classes/HCaptchaWebViewManager.swift b/HCaptcha/Classes/HCaptchaWebViewManager.swift
index 06c0606..ba7d48c 100644
--- a/HCaptcha/Classes/HCaptchaWebViewManager.swift
+++ b/HCaptcha/Classes/HCaptchaWebViewManager.swift
@@ -141,6 +141,9 @@ internal class HCaptchaWebViewManager {
         }
 #endif
         view.addSubview(webView)
+        if webView.bounds.size == CGSize.zero {
+            self.configureWebView?(self.webView)
+        }
 
         executeJS(command: .execute)
     }
```

**Comment by e271828-** [comment]
@CAMOBAP seems plausible, and we should certainly add the example code as a test.

**Comment by e271828-** [comment]
Closed in #94

**Comment by muthenberg** [comment]
Thanks for the effort. I hope it gets released soon. :D
